CMakeLists.txt

```diff
@@ -53,14 +55,31 @@ if(WITH_XC_HTTP)
   set(WITH_XC_NETWORKING ON CACHE BOOL "Include networking code (e.g. for downlading website icons)." FORCE)
 endif()
 
+if(WITH_XC_ALL)
+  # Enable all options
+  set(WITH_XC_AUTOTYPE ON)
+  set(WITH_XC_NETWORKING ON)
+  set(WITH_XC_BROWSER ON)
+  set(WITH_XC_HTTP ON) # Deprecated
+  set(WITH_XC_YUBIKEY ON)
+  set(WITH_XC_SSHAGENT ON)
+endif()
+
 # Process ui files automatically from source files
 set(CMAKE_AUTOUIC ON)
 
 set(KEEPASSXC_VERSION_MAJOR "2")
-set(KEEPASSXC_VERSION_MINOR "2")
-set(KEEPASSXC_VERSION_PATCH "4")
+set(KEEPASSXC_VERSION_MINOR "3")
+set(KEEPASSXC_VERSION_PATCH "0")
 set(KEEPASSXC_VERSION "${KEEPASSXC_VERSION_MAJOR}.${KEEPASSXC_VERSION_MINOR}.${KEEPASSXC_VERSION_PATCH}")
 
+set(KEEPASSXC_RELEASE_BUILD OFF)
+if(NOT KEEPASSXC_RELEASE_BUILD)
```

Comment:
This does not work. You need to remove `set(KEEPASSXC_RELEASE_BUILD OFF)`.

Reply:
That is the point, before you merge to make a release tag you must set that to ON. This means that tag-based builds will be considered "stable" releases and not receive the warning or the -snapshot tag. When you merge back into develop you set this back to off.

Reply:
No, I mean it does not work *at all*. No matter whether I specify -DKEEPASSXC_RELEASE_BUILD=ON, this will always reset it to false.

Reply:
My only reservation with making it a flag is that our source packages will build as a snapshot if the builder doesnt supply the release flag. I might be over thinking this though, it is easily solved with documentation.

Reply:
Corrected in last push.

Reply:
Could we maybe add a tag check and disable it automatically if HEAD points to a release/* tag?

Reply:
I thought of that, it is not terribly difficult to do

Reply:
Willing to do that? :smiley: 
(release-tool should explicitly set the flag nonetheless)

Reply:
added!